library/core/src/main/java/com/google/android/exoplayer2/upstream/ContentDataSource.java

```diff
@@ -89,7 +89,7 @@ public long open(DataSpec dataSpec) throws ContentDataSourceException {
         long assetFileDescriptorLength = assetFileDescriptor.getLength();
         if (assetFileDescriptorLength == AssetFileDescriptor.UNKNOWN_LENGTH) {
           // The asset must extend to the end of the file.
-          bytesRemaining = inputStream.available();
+          bytesRemaining = ((FileInputStream) inputStream).getChannel().size();
```

Comment:
I think it's possible for multiple assets to be in a single file. In which case is `size()` the size of just this one, or of all of them? I think you might want to subtract `position()` of the `FileChannel` to get correct behavior?

Reply:
Not sure what do you mean, FileChannel one per file.

Reply:
I don't think the `uri` for which the source is being opened necessarily corresponds to the whole file though. Else this code wouldn't need to be using `assetFileDescriptor.getStartOffset` as above? If the `uri` only corresponds to a part of the file, that needs to be accounted for.

Reply:
Even if the `uri` does correspond to the whole file, you'd need to subtract `skipped` if it's non-zero.

Reply:
Then correct line is:

bytesRemaining = ((FileInputStream) inputStream).getChannel().size() - assetFileDescriptor.getStartOffset